curvefs/src/metaserver/partition.cpp

```diff
@@ -0,0 +1,148 @@
+    assert(paritionInfo.start() <= paritionInfo.end());
+
+    inodeStorage_ = std::make_shared<MemoryInodeStorage>();
+    dentryStorage_ = std::make_shared<MemoryDentryStorage>();
+    inodeManager_ = std::make_shared<InodeManager>(inodeStorage_);
+    dentryManager_ = std::make_shared<DentryManager>(dentryStorage_);
+    partitionInfo_ = paritionInfo;
+    if (!paritionInfo.has_nextid()) {
+        partitionInfo_.set_nextid(partitionInfo_.start());
+    }
+}
+
+// dentry
+MetaStatusCode Partition::CreateDentry(const Dentry& dentry) {
+    return dentryManager_->CreateDentry(dentry);
+}
+
+MetaStatusCode Partition::GetDentry(uint32_t fsId, uint64_t parentId,
+                                    const std::string& name, Dentry* dentry) {
+    return dentryManager_->GetDentry(fsId, parentId, name, dentry);
+}
+
+MetaStatusCode Partition::DeleteDentry(uint32_t fsId, uint64_t parentId,
+                                       const std::string& name) {
+    return dentryManager_->DeleteDentry(fsId, parentId, name);
+}
+
+MetaStatusCode Partition::ListDentry(uint32_t fsId, uint64_t dirId,
+                                     std::list<Dentry>* dentryList) {
+    return dentryManager_->ListDentry(fsId, dirId, dentryList);
+}
+
+// inode
+MetaStatusCode Partition::CreateInode(uint32_t fsId, uint64_t length,
+                                      uint32_t uid, uint32_t gid, uint32_t mode,
+                                      FsFileType type,
+                                      const std::string& symlink,
+                                      Inode* inode) {
+    uint64_t inodeId = GetNewInodeId();
+    if (inodeId == UINT64_MAX) {
+        return MetaStatusCode::PARTITION_ALLOC_ID_FAIL;
+    }
+    return inodeManager_->CreateInode(fsId, inodeId, length, uid, gid, mode,
+                                      type, symlink, inode);
+}
+
+MetaStatusCode Partition::CreateRootInode(uint32_t fsId, uint32_t uid,
+                                          uint32_t gid, uint32_t mode) {
+    return inodeManager_->CreateRootInode(fsId, uid, gid, mode);
+}
+
+MetaStatusCode Partition::GetInode(uint32_t fsId, uint64_t inodeId,
+                                   Inode* inode) {
+    return inodeManager_->GetInode(fsId, inodeId, inode);
+}
+
+MetaStatusCode Partition::DeleteInode(uint32_t fsId, uint64_t inodeId) {
+    return inodeManager_->DeleteInode(fsId, inodeId);
+}
+
+MetaStatusCode Partition::UpdateInode(const Inode& inode) {
+    return inodeManager_->UpdateInode(inode);
+}
+
+MetaStatusCode Partition::UpdateInodeVersion(uint32_t fsId, uint64_t inodeId,
+                                             uint64_t* version) {
+    return inodeManager_->UpdateInodeVersion(fsId, inodeId, version);
+}
+
+MetaStatusCode Partition::InsertInode(const Inode& inode) {
+    return inodeManager_->InsertInode(inode);
+}
+
+bool Partition::IsDeletable() {
+    // if patition has no inode or no dentry, it is deletable
```

Comment:
maybe it better move this annotation to function declaration

Reply:
done